sphinx/directives/patches.py

```diff
@@ -55,9 +56,45 @@ def run(self):
         return result
 
 
+class RSTTable(tables.RSTTable):
+    """The table directive which sets source and line information to its caption."""
+
+    def make_title(self):
+        title, message = tables.RSTTable.make_title(self)
+        if title:
+            set_source_info(self, title)
+
+        return title, message
```

Comment:
Until now, I patched the problem part at sphinx.util.nodes.apply_source_workaround, and deleted the patch after modifying docutils side. I'm worried that the inherited class implementation can be used for purposes other than workaround.